Source/WebCore/style/RuleSet.h

```diff
@@ -119,8 +119,15 @@ class RuleSet : public RefCounted<RuleSet> {
     bool hasContainerQueries() const { return !m_containerQueries.isEmpty(); }
     Vector<const CQ::ContainerQuery*> containerQueriesFor(const RuleData&) const;
 
+    using ScopeRuleIdentifier = unsigned;
+    struct ScopeAndParent {
+        Ref<const StyleRuleScope> scopeRule;
+        ScopeRuleIdentifier parent;
+        RefPtr<const ContainerNode> ownerNode;
+    };
```

Comment:
I know I suggested this approach but I now think there is actually a problem with putting Node pointers into RuleSets. The issue is that we can share Resolvers (and so RuleSets) between different shadow trees (`Scope::createOrFindSharedShadowTreeResolver()`). While this sharing could be disallowed in this case it would probably be architecturally better to not have Nodes here.

The alternative strategy of adding StyleSheetContents pointer to StyleRuleScope and find the owner node(s) via it (and either disabling caching in `StyleSheetContents::isCacheable()` or iterating over clients to find the relevant owners) would probably be cleaner and less code.